packages/public-api/src/packageInfo/packageSummaryService.ts

```diff
@@ -0,0 +1,73 @@
+import { getRepository } from 'typeorm';
+import {
+  PackageMetadata,
+  PackageUsageByHostnameProjection,
+  PackagePopularityView,
+  PackageVulnerability,
+  toSerializable,
+} from '@gradejs-public/shared';
+import semver from 'semver';
+
+export async function getPackageSummaryByName(packageName: string) {
+  const packageRepo = getRepository(PackageMetadata);
+  const packageUsageRepo = getRepository(PackageUsageByHostnameProjection);
+  const packagePopularityRepo = getRepository(PackagePopularityView);
+  const packageVulnerabilities = getRepository(PackageVulnerability);
+
+  const queries = [
+    packageRepo.createQueryBuilder().where('name = :packageName', { packageName }),
+    packageUsageRepo
+      .createQueryBuilder('usage')
+      .leftJoinAndSelect('usage.hostname', 'hostname')
+      .where('package_name = :packageName', { packageName })
+      .limit(16), // TODO: remove hardcode
+    packagePopularityRepo
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+    packageVulnerabilities
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+  ];
+
+  const [[packageInfo], usage, [popularity], vulnerabilities] = (await Promise.all(
+    queries.map((q) => q.getMany())
+  )) as [
+    PackageMetadata[],
+    PackageUsageByHostnameProjection[],
+    PackagePopularityView[],
+    PackageVulnerability[]
+  ];
+
+  if (!packageInfo) {
+    return null;
+  }
+
+  return {
+    ...toSerializable(packageInfo),
+    updateDate: packageInfo.versionSpecificValues?.[packageInfo.latestVersion]?.updateDate,
+    deps: Object.keys(
+      packageInfo.versionSpecificValues?.[packageInfo.latestVersion]?.dependencies ?? {}
+    ),
+    versionSpecificValues: Object.fromEntries(
+      Object.entries(packageInfo.versionSpecificValues ?? {})
+        .filter(
+          // remove non-release versions from output, keep the latest one
+          ([version]) => packageInfo.latestVersion === version || version.match(/^\d+\.\d+\.\d+$/)
+        )
+        .map(([version, data]) => [
+          version,
+          {
+            ...data,
+            isVulnerable: vulnerabilities.some((v) =>
```

Comment:
it seems like we can send a flat list of `vulnerabilities` and `popularity` to the front-end and transform the data on the front-end side